### DISCUSSION TITLE
No field changes registered when using null as default value

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [react-hook-form/react-hook-form#10636] [Q&A] [answered]
Hey folks,

I recently came across some weird behaviour when using null as a default value of a field.
I've prepared a codesandbox for anyone who wants to take a look at it: [CodeSandbox](https://codesandbox.io/s/primereact-test-forked-s82lh9?file=/src/Form1.jsx)

In this form I have a dropdown and a MultiSelect, the availability of the second one depends on the value of the first one, if the first one has nothing selected, the second one is disabled.

Now the first time I select a value in the first dropdown the behaviour is correct, and the availability of the MultiSelect changes.
But now when I click on the clear icon of the first dropdown, the dropdown is cleared but no change is detected in the field.

If I specify undefined as defaultValue everything works correctly, so that would mean that !getValues("dd1") compares the current value with the default and not whether there is a falsy value behind it like null or undefined.

In the codesandbox you can see i have the following component:
```jsx
<Dropdown disabled={!getValues("dd1")}/>
```

And in javascript, both `!null` and `!undefined!` are equal to `true`, so i wonder why the disabled state is only updated when i use undefined as default value and noto null, since both should have the same effect:

Reproduction steps:

1. Open the console in Codesandbox to see the field value changes getting logged
2. Click the open panel button
3. See that the second dropdown is disabled
4. Select something from the first dropdown
5. Check the console to see that the field value has changed
6. See that dropdown2 is now enabled
7. Now clear dropdown1
8. Dropdown2 stays enabled
9. Select something from dropdown1
10. Notice that no field changes are logged anymore
11. (Change the defaultValue of `dd1` to undefined and everything will work)

I'm not creating a bug report sind I'm not sure if this a bug or I'm not understanding something correctly.
Therefore I hope someone can explain this behaviour to me.

Thanks in advance!

**Comment by tparvi** [comment, marked as answer]
Modified your code to use `useWatch` instead of `getValues` https://codesandbox.io/s/primereact-test-forked-g8jmph?file=/src/Form1.jsx

I would not recommend using `getValues` unless you have e.g. save button and you want to read the whole form when someone clicks it. Also see documentation: https://www.react-hook-form.com/api/useform/getvalues/

_An optimized helper for reading form values. The difference between watch and getValues is that getValues will not trigger re-renders or subscribe to input changes._

**Reply by manuelpoelzl** [reply]
Wow, I must have completely overlooked that possibility.

Thank you for the sample and the link to the documentation!
